### ISSUE TITLE
Insert tab before cursor

### ISSUE THREAD

**Issue by dewyze** [RRethy/nvim-treesitter-endwise#8]
If you are not using tree-sitter, when you press <Enter> after a ruby class/if/def etc. will indent the cursor (`|`): 

```ruby
it "works" do
  |
end
```

Currently with tree-sitter, that does not work. However, I think we could make it work with this plugin.

I was able to get it to work by adding:

```lua
vim.api.nvim_put({ string.rep(' ', vim.api.nvim_buf_get_option(0, 'tabstop')) }, 'c', true, true)
```

Right after this line: https://github.com/RRethy/nvim-treesitter-endwise/blob/f7e0ad42a58c2da1fcd654a1394e9d79f52d6c53/lua/nvim-treesitter/endwise.lua#L64

I am not sure that is the most lua-y way to solve this, but what are your thoughts on adding something like that?

**Comment by RRethy** [comment]
Are you using `indent = { enable = true }` in your tree-sitter config? That won't work with endwise because we add the text after the `<cr>` gets evaluated (and thus after the indent calculations).

I used to have something similar to that ([here](https://github.com/RRethy/nvim-treesitter-endwise/commit/83fc234c378c4bc337171bd3ed9a4bcbdb846d6e#diff-ae3c1217054f2917ebd362bb61d679ba72ac41c6d42e6293e2cb3b77e7687606R14-R20) and [here](https://github.com/RRethy/nvim-treesitter-endwise/commit/83fc234c378c4bc337171bd3ed9a4bcbdb846d6e#diff-ae3c1217054f2917ebd362bb61d679ba72ac41c6d42e6293e2cb3b77e7687606R62-R69)), this was back when I was using a [mapping](https://github.com/RRethy/nvim-treesitter-endwise/commit/83fc234c378c4bc337171bd3ed9a4bcbdb846d6e#diff-ae3c1217054f2917ebd362bb61d679ba72ac41c6d42e6293e2cb3b77e7687606R133-R138) instead of `vim.on_key`. I removed it because it's really hard to write that type of code, your example will break for filetypes with varying values of `'shiftwidth'`, `'expandtab'`, `'vartabstop'`, and it will break for languages like Lua which tree-sitter calculates indentation differently than ruby. The code I linked will break with different values for `'vartabstop'`. There isn't a great solution for this, even `indent.c` in the Neovim source is decently complicated.

I think the best solution might be to go through FFI and call a Neovim C function to get the correct indentation, will have to investigate a bit more though. Personally I don't use tree-sitter's indenting because I found it super buggy even without endwise, right now I'd recommend avoiding tree-sitter based indenting.
